### PR TITLE
Keystone: Use Yaodu Ubuntu Image

### PR DESCRIPTION
**Pull request by intlabs**
This PR proposes the use of the Keystone Image from the Yaodu Project.

For more information about the image please refer to:
 * https://hub.docker.com/r/yaodu/keystone/
 * https://github.com/yaodu/docker-keystone

---
This change is [Reviewable](https://reviewable.io/reviews/att-comdev/openstack-helm/115)

### CODE REVIEW

**Comment by stannum-l** [comment]
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[keystone/templates/configmap-etc.yaml, line 17 at r2](https://reviewable.io:443/reviews/att-comdev/openstack-helm/115#-KaYu6vkso3FijLG2ddy:-KaYu6vkso3FijLG2ddz:bsy8k6i) ([raw file](https://github.com/att-comdev/openstack-helm/blob/17d332e985fcd2695e392c83918b12f9221a5499/keystone/templates/configmap-etc.yaml#L17)):*
> ```YAML
> {{ tuple "etc/_keystone-paste.ini.tpl" . | include "template" | indent 4 }}
>   sso_callback_template.html: |+
> {{ tuple "etc/_keystone-paste.ini.tpl" . | include "template" | indent 4 }}
> ```

should this read `etc/_sso_callback_template.html.tpl`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/att-comdev/openstack-helm/115)*

**Comment by intlabs** [comment]
@stannum-l Good catch, whoops! will get fixes asap/

File: keystone/values.yaml [hunk 1]
```diff
@@ -10,8 +10,8 @@ labels:
   node_selector_value: enabled
 
 images:
-  db_sync: quay.io/stackanetes/stackanetes-keystone-api:newton
-  api: quay.io/stackanetes/stackanetes-keystone-api:newton
+  db_sync: docker.io/yaodu/keystone:ubuntu
```
Comment (alanmeadows) [inline keystone/values.yaml]: How will this allow the user to target a particular OpenStack release? ```:newton``` vs ```:ubuntu```

File: keystone/templates/etc/_keystone-paste.ini.tpl [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+[pipeline:public_version_api]
+pipeline = healthcheck cors sizelimit osprofiler url_normalize public_version_service
+
+[pipeline:admin_version_api]
+pipeline = healthcheck cors sizelimit osprofiler url_normalize admin_version_service
+
+[composite:main]
+use = egg:Paste#urlmap
+/v2.0 = public_api
```
Comment (jallirs) [inline keystone/templates/etc/_keystone-paste.ini.tpl]: Don't you probably want this set to v3 by default? I thought v2 was depreciated?

File: keystone/templates/etc/_keystone-paste.ini.tpl [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+/v3 = api_v3
+/ = public_version_api
+
+[composite:admin]
+use = egg:Paste#urlmap
+/v2.0 = admin_api
```
Comment (jallirs) [inline keystone/templates/etc/_keystone-paste.ini.tpl]: Same here?

File: keystone/templates/etc/_keystone-paste.ini.tpl [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+[filter:build_auth_context]
+use = egg:keystone#build_auth_context
+
+[filter:token_auth]
+use = egg:keystone#token_auth
+
+[filter:admin_token_auth]
+# This is deprecated in the M release and will be removed in the O release.
+# Use `keystone-manage bootstrap` and remove this from the pipelines below.
+use = egg:keystone#admin_token_auth
```
Comment (jallirs) [inline keystone/templates/etc/_keystone-paste.ini.tpl]: I don't think this is needed anymore. May want to remove admin_token auth.

File: keystone/values.yaml (hunk 1 above)
Comment (v1k0d3n) [inline keystone/values.yaml]: +1

i think you've since thought about this a bit @intlabs, but let us know what your thoughts are on this one.

**Comment by intlabs** [comment]
Closing for now, I'll be back pushing these Images again when both projects are at a level of maturity that makes sense to do so :)

**Pull request closed without merging**